sim/dex-data.ts

```diff
@@ -39,6 +39,21 @@ export function toID(text: any): ID {
 	return _toIDCache.get(text) || text.toLowerCase().replace(/[^a-z0-9]+/g, '') as ID;
 }
 
+/**
+ * Like Object.assign but only fields missing from self.
+ * Facilitates consistent field ordering in constructors.
+ * Modifies self in-place.
+ */
+export function assignNewFields(self: AnyObject, data: AnyObject) {
+	for (const k in data) {
+		// to behave like Object.assign, skip inherited fields
+		if (!data.hasOwnProperty(k)) continue;
+		// FIXME: decide whether this should be 'in' or 'hasOwnProperty'
+		if (k in self) continue;
```

Comment:
Checking own properties should be enough, right? Also, ``Object.hasOwn()`` instead of ``hasOwnProperty``, just in case ``data`` has null prototype.

Reply:
I did try `Object.hasOwn` at first; `tsc` gives 
```
sim/dex-data.ts:51:15 - error TS2550: Property 'hasOwn' does not exist on type 'ObjectConstructor'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2022' or later.

51   if (!Object.hasOwn(data, k)) continue;
```
and MDN says `Object.hasOwn` is in Node 16.9.0+.

As for the checks, I was going off the matrix here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Enumerability_and_ownership_of_properties .

I'm not sure if the `k in self` line should respect existing inherited props or not. I think it should (the current behavior).

Reply:
> Object.hasOwn snip

Might as well go with good old ``Object.hasOwnProperty.call(...)``. I am a supporter of ``Reflect.hasOwnProperty`` myself, but others seem to hate Reflect!

> I'm not sure if the k in self line should respect existing inherited props or not. I think it should (the current behavior).

Poll goes 1-1 so far /shrug